legend-pure-core/legend-pure-m4/src/main/java/org/finos/legend/pure/m4/coreinstance/primitive/date/PureDate.java

```
@@ -77,6 +78,11 @@ public interface PureDate extends Comparable<PureDate>
      */
     GregorianCalendar getCalendar();
 
+    default Instant toInstant()
+    {
+        return this.getCalendar().toZonedDateTime().toInstant();
+    }
+
```

Not all PureDates can be converted to Instants in a valid way. Really, only those with nanosecond or better precision can. Is this method really necessary on this interface?

If it is really necessary, it should be accompanied by another method to test whether the PureDate can validly be converted to an Instant. And the default method should throw UnsupportedOperationException if it can't.

---

addressed in another PR - https://github.com/finos/legend-pure/pull/943/files ; closing this PR